From 01674f6bc6ded19bafe9787159352837894410c0 Mon Sep 17 00:00:00 2001
From: Hongcheng Xie <hongcheng.xie@intel.com>
Date: Fri, 3 Dec 2021 15:33:37 +0800
Subject: [PATCH] Allow override local prop in user version

AIC will override some prop such as virtual.input.num
..etc in data/local.prop, but init process only allow
to override prop in local.prop for non-user version.

Allow override local prop in user version too.

Change-Id: Ibd56a6b8a5675dbc515b7f8a43d7eecf2084510c
Tracked-On: OAM-100218
Signed-off-by: Hongcheng Xie <hongcheng.xie@intel.com>
---
 init/property_service.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/init/property_service.cpp b/init/property_service.cpp
index a89504e59..1bb0d29d3 100644
--- a/init/property_service.cpp
+++ b/init/property_service.cpp
@@ -757,8 +757,10 @@ static void update_sys_usb_config() {
 }
 
 static void load_override_properties() {
-    if (ALLOW_LOCAL_PROP_OVERRIDE) {
+    bool allow_override_prop = android::base::GetBoolProperty("ro.allow.override.prop", true);
+    if (ALLOW_LOCAL_PROP_OVERRIDE || allow_override_prop) {
         std::map<std::string, std::string> properties;
+        LOG(INFO) << "Loading prop from local.prop";
         load_properties_from_file("/data/local.prop", nullptr, &properties);
         for (const auto& [name, value] : properties) {
             std::string error;
-- 
2.39.0

